app/src/hooks/useStorageFolders.ts

```diff
@@ -0,0 +1,69 @@
+import { useState, useCallback } from 'react';
+import { isEmpty } from 'es-toolkit/compat';
+import { isNumber } from 'es-toolkit/compat';
+
+import { EXPANDED_FOLDERS_IDS } from 'common/constants/localStorageKeys';
+import { getStorageItem, setStorageItem } from 'common/utils/storageUtils';
+import { INSTANCE_KEYS } from 'pages/inside/common/expandedOptions/folder/useFolderTooltipItems';
+import { TransformedFolder } from 'controllers/testCase';
+
+const BASE_STORAGE_KEY = EXPANDED_FOLDERS_IDS as string;
+
+const getFolderAndDescendantIds = (folder: TransformedFolder): number[] => {
+  const ids = [folder.id];
+
+  if (!isEmpty(folder.folders)) {
+    folder.folders.forEach((subFolder) => {
+      ids.push(...getFolderAndDescendantIds(subFolder));
+    });
+  }
+
+  return ids;
+};
+
+export const useStorageFolders = (instanceKey?: INSTANCE_KEYS) => {
+  const storageKey = instanceKey ? `${BASE_STORAGE_KEY}_${instanceKey}` : BASE_STORAGE_KEY;
```

Comment:
```suggestion
  const storageKey = instanceKey ? `${EXPANDED_FOLDERS_IDS}_${instanceKey}` : EXPANDED_FOLDERS_IDS;
```

so we can remove `BASE_STORAGE_KEY`